### PR TITLE
Support accessing InfluxDB via HAProxy

### PR DESCRIPTION
**Pull request by dougszumski**

### CODE REVIEW

File: ansible/roles/haproxy/templates/haproxy.cfg.j2 [hunk 1]
```diff
@@ -124,6 +124,20 @@ listen glance_api_external
 {% endif %}
 {% endif %}
 
+{% if enable_influxdb | bool %}
```
Comment (markgoddard) [inline ansible/roles/haproxy/templates/haproxy.cfg.j2]: Why isn't this done upstream? Is this to support a clustered Influx DB? Does that work with the free version now?

Comment (dougszumski) [reply ansible/roles/haproxy/templates/haproxy.cfg.j2]: I can't see an obvious reason why it wasn't done upstream - some other things like Murano were also missed out and added under bug fixes. Conceivably someone could override the Dockerfile to install the enterprise version to support clustering. Without clustering it's still nice to have it behind HA proxy to reduce the configuration burden if InfluxDB was ever moved to a different host. What do you think? I was about to submit this upstream.

Comment (dougszumski) [reply ansible/roles/haproxy/templates/haproxy.cfg.j2]: The InfluxDB Relay looks interesting to support HA without the enterprise version:

https://docs.openstack.org/developer/performance-docs/methodologies/monitoring/influxha.html#influxdb-ha-mechanism-realization

Comment (yankcrime) [reply ansible/roles/haproxy/templates/haproxy.cfg.j2]: SSL termination is also handy.

Comment (dougszumski) [reply ansible/roles/haproxy/templates/haproxy.cfg.j2]: It's up for further debate here: https://review.openstack.org/#/c/547436/

Comment (markgoddard) [reply ansible/roles/haproxy/templates/haproxy.cfg.j2]: Seems reasonable. We won't get SSL termination though as we'll only want to expose this on the internal network.

**Pull request merged**